src/values.rs

```diff
@@ -108,6 +108,8 @@ pub enum ContainerValue<T> {
     Sum(usize, Box<T>), // Tag and value
 }
 
+impl<Elem: Eq> Eq for ContainerValue<Elem> {}
```

Comment:
👍

Reply:
In fact we can just `#[derive(Eq)]` - the `derive` macro is smart enough to condition the generated `impl` on the appropriate condition of the arguments, which is pretty neat.